kong/plugins/datadog/schema.lua

```diff
@@ -101,10 +103,31 @@ return {
                     if_match = { one_of = { "counter", "gauge" }, },
                     then_field = "sample_rate",
                     then_match = { required = true },
-          }, }, }, }, }, },
+                  }, }, }, }, },
+          },
+        },
+
+        entity_checks = {
+          { custom_entity_check = {
+              field_sources = { "retry_count", "queue_size", "flush_timeout" },
+              fn = function(entity)
+                if entity.retry_count and entity.retry_count ~= 10 then
```

Comment:
Maybe a stupid question due to ignorance: since the deprecated fields don't have a default value anymore, why is this deprecation message gated on the older default value? I suspect this may be to address the hybrid deployment (?), but this way a non-hybrid user setting `retry_count=10` will not get the deprecation message logged.

What am I missing?

Reply:
We gate the warning through the old default values is that we can end up in the situation that the defaults were persisted without the user ever specifying them explicitly.  We cannot distinguish between that case and a user setting a parameter to the default value explicitly.